### PR TITLE
Shim interface rendering surface for testing

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
Adds in a shim so that we work with an interface in the Instrument view
instead of a concrete type.

This is the lowest impact way of prepping for tests, if the hierarchy is
wrong it will fail to build of open at all. It also means we can start
building tests without touching the implementation from this point
onwards.

One unfortunate problem is the fact the interfaces inherit from Qt
types, this is the least impact way to get this working through.

**To test:**
- Load some data, I find POLARIS data has one of the clearest geometry 
- Open instrument view, switch between the various types of render

Part of #31763 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by gemmaguest — APPROVED** [review]
Looks good, still works as expected :+1:

No release notes required because this is internals.

File: qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/SimpleWidget.h [hunk 1]
```diff
@@ -18,21 +20,21 @@ class ProjectionSurface;
 /**
  * A simple widget for drawing unwrapped instrument images.
  */
-class SimpleWidget : public QWidget {
+class SimpleWidget final : public ISimpleWidget {
```
Comment (martyngigg) [inline qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/SimpleWidget.h:23]: I know you didn't pick this name but now we're refactoring could we think of a better one? `Simple` conveys so little information.

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/SimpleWidget.h (hunk 1 above)
Comment (gemmaguest) [reply qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/SimpleWidget.h:23]: @martyngigg Sure. Can we do that in a future PR though? We'd like to change as little as possible until we have some tests in.

Comment (gemmaguest) [reply qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/SimpleWidget.h:23]: I've added a comment to #31739 to do this

Comment (martyngigg) [reply qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/SimpleWidget.h:23]: Yeah no problem. Thanks!

**Review by martyngigg — APPROVED** [review]

**Pull request merged**